src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PathSeqFilterSpark.java

```diff
@@ -0,0 +1,192 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.github.lindenb.jbwa.jni.ShortRead;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.common.collect.Iterators;
+import htsjdk.samtools.*;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.ArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.OpticalDuplicatesArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.filters.*;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.tools.spark.sv.SVKmerSmall;
+import org.broadinstitute.hellbender.tools.spark.utils.HopscotchSet;
+import org.broadinstitute.hellbender.transformers.BaseQualityClipReadTransformer;
+import org.broadinstitute.hellbender.transformers.DUSTReadTransformer;
+import org.broadinstitute.hellbender.transformers.ReadTransformer;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import org.broadinstitute.hellbender.utils.read.SAMFileGATKReadWriter;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+import org.broadinstitute.hellbender.utils.read.markduplicates.MarkDuplicatesScoringStrategy;
+import org.broadinstitute.hellbender.utils.read.markduplicates.OpticalDuplicateFinder;
+import org.broadinstitute.hellbender.tools.spark.transforms.markduplicates.MarkDuplicatesSpark;
+import org.seqdoop.hadoop_bam.BAMInputFormat;
+import scala.Tuple2;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(summary = "Read preprocessing and host filtering on reads from a BAM file",
+        oneLineSummary = "PathSeqFilter on Spark",
+        programGroup = SparkProgramGroup.class)
+public final class PathSeqFilterSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+    private SAMFileHeader header;
+
+    private static final String maxLengthArgName = "pathseq_maxReadLength";
+    @Argument(fullName = maxLengthArgName,
```

Comment:
This was just because I was using the existing ReadLengthFilter that requires an upper bound. I've done away with the argument and instead passed Integer.INT_MAX to the filter.